server.ts

```diff
@@ -0,0 +1,81 @@
+// These are important and needed before anything else
+import 'zone.js/dist/zone-node';
+import 'reflect-metadata';
+
+import { enableProdMode } from '@angular/core';
+
+import * as express from 'express';
+import { join } from 'path';
+
+// Faster server renders w/ Prod mode (dev mode never needed)
+enableProdMode();
+
+// Express server
+const app = express();
+
+const PORT = process.env.PORT || 3451;
+const DIST_FOLDER = join(process.cwd());
+const CONFIG = {
+  'api_url': process.env['SNL_API_URL'] || 'http://localhost:8090'
+};
+
+// * NOTE :: leave this as require() since this file is built Dynamically from webpack
+const { AppServerModuleNgFactory, LAZY_MODULE_MAP } = require('./dist/server/main.bundle');
+
+// Express Engine
+import { ngExpressEngine } from '@nguniversal/express-engine';
+// Import module map for lazy loading
+import { provideModuleMap } from '@nguniversal/module-map-ngfactory-loader';
+import { AppConfig } from './src/app/app.config';
+import { AppServerConfig } from './src/app/app.server.config';
+
+app.engine('html', ngExpressEngine({
+  bootstrap: AppServerModuleNgFactory,
+  providers: [
+    provideModuleMap(LAZY_MODULE_MAP),
+    { provide: AppConfig, useValue: new AppServerConfig(CONFIG) },
+  ]
+}));
+
+app.all('/*', function(req, res, next) {
+    res.header('Access-Control-Allow-Origin', '*');
```

Comment:
we should not use asterisk *, the exact value of the url should be injected from configuration. (for localhost it may be *, but not for azure hosted environments).